test/JSON/toJSON.js

```diff
@@ -15,12 +15,141 @@ fnc.prototype.toJSON = function() {
   return this.number.toString();
 }
 
+// test getter - no-enum
+{
+  var obj = {};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: false });
+
+  obj.a = 1;
+  obj.b = 2;
+  obj.d = 4;
+  TEST('{"a":1,"b":2,"d":4}', JSON.stringify(obj));
+}
+
+// test getter - no-enum - delete
+{
+  var obj = {};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: false, configurable: true });
+
+  obj.a = 1;
+  obj.b = 2;
+  delete obj['c']
+  TEST('{"a":1,"b":2}', JSON.stringify(obj));;
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: false });
+  obj.d = 4;
+  TEST('{"a":1,"b":2,"d":4}', JSON.stringify(obj));
+}
+
+// test getter - enum - delete 'em all
+{
+  var obj = {};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true, configurable: true });
+
+  obj.a = 1;
+  obj.b = 2;
+  delete obj['c']
+  TEST('{"a":1,"b":2}', JSON.stringify(obj));;
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+  obj.d = 4;
+  TEST('{"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  obj[9]=19;
+  TEST('{"9":19,"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  delete obj[9]
+  TEST('{"c":3,"a":1,"b":2,"d":4}', JSON.stringify(obj));
+  delete obj['a']
+  TEST('{"c":3,"b":2,"d":4}', JSON.stringify(obj));
+}
+
+// test getter
+{
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+
+  obj.d = 4;
+  TEST('{"a":1,"b":2,"c":3,"d":4}', JSON.stringify(obj));;
+}
+
+// test getter - setter
+{
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true }); // should print
+  Object.defineProperty(obj, "d", { set: () => 4, enumerable: true }); // skips undefined
+
+  obj.e = 5;
+  TEST('{"a":1,"b":2,"c":3,"e":5}', JSON.stringify(obj));
+}
+
+// test getter with revive
+{
+  function rev(name, value) {
+    if (value == 3) {
+      return undefined;
+    }
+    else if (value == true) {
+      return 99;
+    }
+    return value;
+  }
+  var obj = {a:1, b:2};
+  Object.defineProperty(obj, "c", { get: () => 3, enumerable: true });
+
+  obj.d = 4;
+  TEST('{"a":99,"b":2,"d":4}', JSON.stringify( JSON.parse(JSON.stringify(obj), rev) ));
+}
+
+// test getter with revive && objectArray
+{
```

Comment:
Is it possible to mix proxy with JSON.stringify? don't have test handy but changing the state when trap is invoked? although that would be irrelevant if this PR is meant to improve/address perf.

Reply:
proxy is handled in separate code place and I didn't change that part so far. However, i.e. `Date .toString` depends on internal `toJSON` and I had added tests to cover those.